### PR TITLE
Add tests and documentation for <dynamic-array-app> web component

### PR DESCRIPTION
**Pull request by YuriiHaidamaka**
In this PR we have added tests and documentation for <dynamic-array-app> web component.

### CODE REVIEW

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 1]
```diff
@@ -10,44 +10,69 @@
...
-        test('setting a property on the element works', function() {
-          // Create a test fixture
-          var element = fixture('ChangedPropertyTestFixture');
-          assert.equal(element.prop1, 'new-prop1');
-          var elementShadowRoot = element.shadowRoot;
-          var elementHeader = elementShadowRoot.querySelector('h2');
-          assert.equal(elementHeader.innerHTML, 'Hello new-prop1!');
+        test('should add new element in the names array after click on add button', function () {
+            element.inputValue = 'Ivan';
+            var elementShadowRoot = element.shadowRoot;
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: Let's not use `var`. Please refactor with `let` or `const`.

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 2]
```diff
@@ -10,44 +10,69 @@
...
+</test-fixture>
+
+<script>
+    suite('dynamic-array-app', function () {
+        let element;
+        setup(() => {
+            element = fixture('BasicTestFixture');
+        })
+
+        test('should instantiating the element with empty array in properties', function () {
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: Should be `instantiate`.

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 3]
```diff
@@ -10,44 +10,69 @@
...
+            flush(function () {
+                var deleteButton = element.shadowRoot.querySelector('div:nth-child(4) > paper-button');
+                deleteButton.click();
+                assert.equal(element.names.length, 1)
+                assert.equal(element.names[0], "Petr")
+                done();
+            });
+        });
+
+        test('should setting a property on the element works', function () {
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: This test is more about attributes and properties binding. Actually it's the functionality of Polymer.Element. Let's test added functionality only.

File: wct.conf.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "verbose": true,
+  "persistent": true,
+  "plugins": {
+    "local": {
+      "browsers": [
+        "chrome"
```
Comment (dkalinin) [inline wct.conf.json]: What about other browsers?

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 5]
```diff
@@ -19,57 +19,43 @@
...
-            var elementShadowRoot = element.shadowRoot;
-            var paperButton = elementShadowRoot.querySelector('paper-button');
+        test('should instantiate the element with add button', function () {
+            const elementShadowRoot = element.shadowRoot;
+            const paperButton = elementShadowRoot.querySelector('paper-button');
             assert.equal(paperButton.innerHTML, 'Add');
         });
 
         test('should add new element in the names array after click on add button', function () {
             element.inputValue = 'Ivan';
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: If you use `Ivan` twice in a test then it should be a const variable in the context of this test. If you use `Ivan` in other tests then it likely should be a suite constant.

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 6]
```diff
@@ -19,57 +19,43 @@
...
-            var elementShadowRoot = element.shadowRoot;
-            var paperButton = elementShadowRoot.querySelector('paper-button');
+            const elementShadowRoot = element.shadowRoot;
+            const paperButton = elementShadowRoot.querySelector('paper-button');
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: Please move selectors to constants.

**Comment by YuriiHaidamaka** [comment]
@dkalinin  PTAL

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 7]
```diff
@@ -32,26 +34,29 @@
 
         test('should instantiate the element with add button', function () {
             const elementShadowRoot = element.shadowRoot;
-            const paperButton = elementShadowRoot.querySelector('paper-button');
-            assert.equal(paperButton.innerHTML, 'Add');
+            const addButtonSelector = 'paper-button';
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: `const addButtonSelector = 'paper-button';` has a duplicate in other test.

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 8]
```diff
@@ -32,26 +34,29 @@
...
+            element.inputValue = nameIvan;
             const elementShadowRoot = element.shadowRoot;
-            const paperButton = elementShadowRoot.querySelector('paper-button');
-            paperButton.click();
+            const addButtonSelector = 'paper-button';
+            const addButton = elementShadowRoot.querySelector(addButtonSelector);
+            addButton.click();
             const names = element.names;
-            assert.equal(names[0], "Ivan");
+            assert.equal(names[0], nameIvan);
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: It's better to test DOM elements, not properties.

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 9]
```diff
@@ -32,26 +34,29 @@
...
         });
 
         test('should delete element from names array after click on delete button', function (done) {
-            element.names = ['Ivan', 'Petr'];
+            element.names = [namePetr, nameIvan];
             flush(function () {
-                const deleteButton = element.shadowRoot.querySelector('div:nth-child(4) > paper-button');
+                const deleteButtonSelector = 'div:nth-last-child(2) > paper-button';
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: Selectors with `div` are not descriptive and can be easily broken after html structure changes. It's better to use classes, ids, custom element tags. If you add class `name-item` to the div which wraps name text and button the selector will be more obvious: `.name-item:nth-last-child(2) > paper-button`.

Also, don't forget to move selectors to the constants.

**Comment by YuriiHaidamaka** [comment]
@dkalinin  PTAL on the test `should delete element from names array after click on delete button`. How should I use two `flush` functions correctly? This test does not work in the IE.

**Comment by YuriiHaidamaka** [comment]
@dkalinin PTAL

File: wct.conf.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
+{
+  "verbose": true,
+  "persistent": true,
+  "plugins": {
+    "local": {
+      "browsers": [
+        "chrome",
+        "firefox",
+        "ie"
```
Comment (dkalinin) [inline wct.conf.json]: Please also add Safari.

File: test/dynamic-array-app/dynamic-array-app_test.html [hunk 11]
```diff
@@ -10,44 +10,65 @@
...
+
+        test('should delete element from names array after click on delete button', function (done) {
+            element.names = [...TEST_NAMES];
+            const elementShadowRoot = element.shadowRoot;
+            flush(function () {
+                const deleteButton = elementShadowRoot.querySelectorAll(DELETE_BUTTON_SELECTOR)[0];
+                deleteButton.click();
+                flush(function () {
+                    const name = elementShadowRoot.querySelectorAll(NAME_SELECTOR)[0];
+                    assert.equal(name.innerHTML, TEST_NAMES[1]);
```
Comment (dkalinin) [inline test/dynamic-array-app/dynamic-array-app_test.html]: It's obvious to check here that there is the only one `name-item`.

**Pull request merged**